packages/@react-aria/test-utils/src/TableTester.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, fireEvent, waitFor, within} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+interface TableOptions {
+  element: HTMLElement,
+  timerType?: 'fake' | 'real'
+}
+
+// TODO: move to general util file and replace our existing one with this
+async function triggerLongPress(element: HTMLElement, pointerOpts = {}, timerType: 'fake' | 'real' = 'fake', waitTime: number = 500) {
+  fireEvent.pointerDown(element, {pointerType: 'touch', ...pointerOpts});
+  if (timerType === 'real') {
+    await act(async () => await new Promise((resolve) => setTimeout(resolve, waitTime)));
+  } else {
+    act(() => {
+      jest.advanceTimersByTime(waitTime);
+    });
+  }
+
+  fireEvent.pointerUp(element, {pointerType: 'touch', ...pointerOpts});
+}
+
+export class TableTester {
+  element: HTMLElement;
+  timerType: 'fake' | 'real';
+  rows: HTMLElement[];
+  columns: HTMLElement[];
+  private rowgroups: HTMLElement[];
+
+  constructor(opts: TableOptions) {
+    let {
+      element,
+      timerType = 'fake'
+    } = opts;
+    this.element = element;
+    this.timerType = timerType;
+    // This can potentially become stale? Double check, if so replace with getters that refetch the rows and columns
+    this.rowgroups = within(this.element).getAllByRole('rowgroup');
+    this.columns = within(this.rowgroups[0]).getAllByRole('columnheader');
+    this.rows = within(this.rowgroups[1]).getAllByRole('row');
+  }
+
+  toggleRowSelection(opts: {index?: number, text?: string, needsLongPress?: boolean}) {
```

Comment:
one helper function that i think would help (since it's come up a bunch) is a method to select a given cell, though i guess it'd require the column and row indexes or text

Reply:
sounds good, we have a util for that that we use internally as well actually haha. I think we can start off with searching for cell text first since that is what a user would look for typically and see about supporting a (x row, y column) alternative maybe.